packages/block-library/src/post-navigation-link/edit.js

```diff
@@ -94,20 +95,28 @@ export default function PostNavigationLinkEdit( {
 	return (
 		<>
 			<InspectorControls>
-				<PanelBody>
-					<ToggleControl
-						__nextHasNoMarginBottom
+				<ToolsPanel label={ __( 'Settings' ) }>
+					<ToolsPanelItem
+						hasValue={ () => !! showTitle }
 						label={ __( 'Display the title as a link' ) }
-						help={ __(
-							'If you have entered a custom label, it will be prepended before the title.'
-						) }
-						checked={ !! showTitle }
-						onChange={ () =>
-							setAttributes( {
-								showTitle: ! showTitle,
-							} )
+						onDeselect={ () =>
+							setAttributes( { showTitle: false } )
 						}
-					/>
+					>
+						<ToggleControl
+							__nextHasNoMarginBottom
+							label={ __( 'Display the title as a link' ) }
+							help={ __(
+								'If you have entered a custom label, it will be prepended before the title.'
+							) }
+							checked={ !! showTitle }
+							onChange={ () =>
+								setAttributes( {
+									showTitle: ! showTitle,
+								} )
+							}
+						/>
+					</ToolsPanelItem>
 					{ showTitle && (
 						<ToggleControl
```

Comment:
This control also needs to be wrapped in `ToolsPanelItem`

Reply:
You can also use the `ToolsPanelItem` component for conditional control ([Example](https://github.com/WordPress/gutenberg/blob/669662c740bf0e69f58ad1d7e570607c066f31ef/packages/block-library/src/archives/edit.js#L60-L61)).